src/nupic/research/temporal_memory.py

```diff
@@ -340,7 +341,14 @@ def learnOnSegments(self,
     @param predictedInactiveCells       (set)         Indices of predicted inactive cells
     @param prevMatchingSegments         (set)         Indices of segments with
     """
-    for segment in prevActiveSegments | learningSegments:
+    segments = prevActiveSegments | learningSegments
+
+    # Sort segments before iterating for compatibility with C++
+    # Sort with primary key = cell idx, secondary key = segment idx
+    segments = sorted(segments,
```

Comment:
nit: drop `segments` argument to next line with same indentation as `key` argument